doc/source/dev/contributor/quickstart_ubuntu.rst

```diff
@@ -31,7 +32,9 @@ Building SciPy
 
 #. Download, install, and test the latest release of the `Anaconda Distribution of Python`_. In addition to the latest version of Python 3, the Anaconda distribution includes dozens of the most popular Python packages for scientific computing, the Spyder integrated development environment (IDE), the ``conda`` package manager, and tools for managing virtual environments.
 
-   In Ubuntu, the Anaconda download comes as a ``.sh`` file. `In a terminal, navigate <https://help.ubuntu.com/community/UsingTheTerminal>`_ to the directory in which the ``.sh`` file is located and enter ``./file.sh``, where ``file.sh`` is to be replaced with the full name of the Anaconda download. This starts the installation process; from there, simply follow the prompts.
+ `In a terminal, navigate <https://help.ubuntu.com/community/UsingTheTerminal>`_ to the location in which you'd like to install Anaconda. You can download the file using the terminal command ``curl -O URL_OF_FILE``, where ``URL_OF_FILE`` is to be replaced by the URL of the Anaconda installer ``.sh`` file found at the Anaconda Distribution website. When the download is complete, enter ``chmod +x file.sh``, where ``file.sh`` is to be replaced with the full name of the downloaded file. This ensures that the file's permissions allow it to be executed. Next, enter ``./file.sh``, where ``file.sh`` is again to be replaced with the full name of the downloaded file. This starts the installation process. From there, simply follow the prompts, including the "Next Steps" at the end after the installer finishes.
+
+#. `Rename the file`_ ``anaconda3/lib/libgfortran.so`` to ``anaconda3/lib/libgfortran.so_backup``, where ``anaconda3`` is to be replaced with the full path of your Anaconda installation. This file provides an incorrect Fortran shared library; renaming it forces the system to find the right one.
```

Comment:
I definitely recognize this as something that sometimes has to be done. A little unfortunate--I would have thought / hoped there was a way to install the older runtime libraries needed through conda-forge or something like that.

Reply:
*Sometimes* is right. It's been tough to find a set of instructions that will work the same way every time. 

By the time I left Sunday, I had a development version of SciPy working in WSL Ubuntu 18.04 (compiled using Conda's gcc_linux-64, etc...) and in WSL Ubuntu 16.04 (compiled with LinuxBrew gcc). I updated Windows, and afterwards neither worked. I don't remember what the problems were, but I tried a bunch of things and nothing seemed to work, so I decided to start from scratch. 

I couldn't get SciPy to compile in WSL Ubuntu 16.04 with conda's compilers. I ran into the `collect2: error: ld returned 1 exit status` mentioned in gh-8325, but the fixes suggested there didn't work for me.

I was somewhat successful with compilers installed with `apt`. The tests run with one failure from within python (`import scipy; scipy.test()`) but won't run with `runtests.py`; If I remember correctly I get the `undefined symbol: _gfortran_stop_numeric_f08` error during collection.

I'll work on this more after CZI is done.

Reply:
> A little unfortunate--I would have thought / hoped there was a way to install the older runtime libraries needed through conda-forge or something like that.

Agreed, though if it's only modifying a file within the development environment, that's not as bad.  The original instructions said to modify a file in the base environment.

Reply:
> Agreed, though if it's only modifying a file within the development environment, that's not as bad. The original instructions said to modify a file in the base environment.

Isn't it both now?

Reply:
> Isn't it both now?

Yes. According to the instructions, we build SciPy first in the base environment to keep the number of instructions from step 0 to an initial run of the test suite just a bit smaller. We need to rename the file in the base environment to do that. Then we create the virtual environment and rename the file there, too.

Technically, we could skip all the steps that are performed in the base environment, and only build SciPy in the development environment, and then we'd only need to modify the file in the development environment. 

Besides (slightly) reducing the number of instructions from step 0 to running the test suite, another slight advantage of the current approach is that it introduces virtual environments separately from the main process of building SciPy, hopefully giving the beginner a better sense of what the individual steps are really doing. That was part of the original intent, anyway. In a future revision, we could reconsider this approach, but it should probably also be changed for the macOS instructions, and then the tutorial would diverge a bit more from the video.

Reply:
> Yes. According to the instructions, we first build SciPy in the base environment. We need to rename the file in the base environment to do that

This is very unhealthy, please don't recommend that. Having to do it in an env other than `base` is bad enough, but don't mess with `base`.

> Technically, we could skip all the steps that are performed in the base environment, and only build SciPy in the development environment, and then we'd only need to modify the file in the development environment.

+1

Reply:
> I have not changed the instructions to avoid having to alter the conda base environment libgfortran.so file. To keep the diff readable, can I do that in a second PR?

Another PR to get the renaming stuff correct is fine, but I would just get rid of the sections about renaming entirely here in that case.

But really I'm not too worried about the diff becoming worse, it's not so bad currently. Clicking the "hide whitespace changes" makes it even more manageable.

Reply:
OK. I'll try to change it tonight.